### ISSUE TITLE
Enable custom uris for feature id for geojson providers

### ISSUE THREAD

**Issue by ksonda** [geopython/pygeoapi#433]
**Is your feature request related to a problem? Please describe.**
I would like to have the id:/ @id: field in the json-ld content for feature collections generated from geojson providers and their constituent items have custom uri (as provided by an external PID service). 

For example: the persistent identifier [https://geoconnex.us/ref/nat_aq/N100BSNRGB](https://geoconnex.us/ref/nat_aq/N100BSNRGB) redirects to [https://info.geoconnex.us/collections/nat_aq/items/N100BSNRGB](https://info.geoconnex.us/collections/nat_aq/items/N100BSNRGB)

In the json-ld representation of this item [https://info.geoconnex.us/collections/nat_aq/items/N100BSNRGB](https://info.geoconnex.us/collections/nat_aq/items/N100BSNRGB?f=jsonld), there is

```
id: "https://info.geoconnex.us/collections/nat_aq/items/N100BSNRGB"
```

I would like this to be the persistent identifier instead.

**Describe the solution you'd like**
One possible solution would be to allow for a ```uri_field:``` parameter in the configuration .yml, which if provided, would override the json-ld generation occuring around line 178 of [linked_data.py](https://github.com/geopython/pygeoapi/blob/master/pygeoapi/linked_data.py) with the entry in the desingated field of the geojson file. 

Of course, this could be generalized to other provider types as well.

Ideally, this would propogate to everywhere in the API and html templates wherever "id" has a hyperlink pointing to its items.html instance.

**Describe alternatives you've considered**
Another solution would be to allow for a ```uri_stem:``` parameter for a resource that would replace ```{}/collections/``` and remove ```items/``` from the ```id:``` generated in linked_data.py. For example, I could specify:

```
resources:
    nat_aq:
        type: collection
        title: National Aquifers
        description: National Aquifers of the United States
        uri_stem: https://geoconnex.us/ref/nat_aq/

```

**Comment by alpha-beta-soup** [comment]
Without commending on implementation details, I'm totally in agreement that:
- the @id should default to being a link to the item (i.e. a direct link to the pygeoapi instance)
- it should be possible to specify a property that can override this (as a literal and maybe as a template?) to support PIDs as well as the semantic meaning of @id

This is what I should have implmented to begin with, I struggled with what to do with @id by default.

**Comment by tomkralidis** [comment]
Can this be closed given #676 ?

**Comment by ksonda** [comment]
Yep!
